concept/type/impl/RelationTypeImpl.java

```diff
@@ -81,9 +81,13 @@ public void setLabel(String label) {
 
     @Override
     public void setAbstract() {
-        if (getInstances().first().isPresent()) throw exception(TypeDBException.of(TYPE_HAS_INSTANCES_SET_ABSTRACT, getLabel()));
-        vertex.isAbstract(true);
-        declaredRoles().forEachRemaining(RoleTypeImpl::setAbstract);
+        if (!isAbstract()) {
```

Comment:
this is just for cleanliness - setting abstract should do nothing (eg. be idempotent) if already abstract. We technically don't need this check, but it leads to fewer cases to think about in the future if we do -- the validation we run is only scoped to when it is not abstract, we don't have to consider it if it is. It also feels more correct to do the validation for instances when not abstract.